lib/mix/lib/mix/utils.ex

```diff
@@ -775,6 +765,11 @@ defmodule Mix.Utils do
           :httpc.set_options([ipfamily: fallback(inet)], :mix)
           request |> httpc_request(http_options) |> httpc_response()
 
+        {:error, {:failed_connect, [{:to_address, _}, {inet, _, {:tls_alert, _}}]}}
+        when inet in [:inet, :inet6] ->
```

Comment:
Maybe
```suggestion
      {:error, {:failed_connect, [{:to_address, _}, {inet, _, reason}]}}
        when inet in [:inet, :inet6] and elem(reason, 0) == :tls_alert ->
```
or something like that?

Relevant:
- https://github.com/erlang/otp/blob/5187d4e3cf435f9c66dab369123f993dda6221fb/lib/ssl/src/ssl.erl#L571-L577
- https://github.com/search?q=repo%3Aerlang%2Fotp%20tls_alert&type=code

Reply:
I think less strict is good as it is plausible to add more elements to the tuple.